src/librustc/mir/interpret/allocation.rs

```diff
@@ -22,6 +22,26 @@ pub enum InboundsCheck {
     MaybeDead,
 }
 
+/// Used by `check_in_alloc` to indicate context of check
+#[derive(Debug, Copy, Clone, RustcEncodable, RustcDecodable, HashStable)]
+pub enum CheckInAllocMsg {
+    MemoryAccess,
+    NullPointer,
+    PointerArithmetic,
+    OutOfBounds,
```

Comment:
This is the only case that is "negatively named", i.e. it describes the error, not the operation that was performed and failed. "InboundsTest" might be a better name?

Reply:
fixed